src/NHibernate/Proxy/Poco/BasicLazyInitializer.cs

```diff
@@ -81,10 +81,6 @@ public virtual object Invoke(MethodInfo method, object[] args, object proxy)
 				{
 					return null;
 				}
-				else if ("get_HibernateLazyInitializer".Equals(methodName))
```

Comment:
We, probably, need to revert this change if we want to support [legacy] proxy factories.

Reply:
Yes, and it looks to me there is no easy way to obsolete this code path: I do not think we can know there if the property implementation is itself proxied or not.